core/connection/manager.go

```diff
@@ -137,19 +144,62 @@ func (manager *connectionManager) Connect(consumerID identity.Identity, proposal
 
 	sessionDTO, paymentInfo, err := manager.createSession(connection, dialog, consumerID, proposal)
 	if err != nil {
+		manager.sendSessionStatus(dialog, "", connectivity.StatusSessionEstablishmentFailed, err)
 		return err
 	}
 
 	err = manager.launchPayments(paymentInfo, dialog, consumerID, providerID)
 	if err != nil {
+		manager.sendSessionStatus(dialog, sessionDTO.ID, connectivity.StatusSessionPaymentsFailed, err)
 		return err
 	}
 
+	currentPublicIP := manager.getPublicIP()
+	// Try to establish connection with peer.
 	err = manager.startConnection(connection, consumerID, proposal, params, sessionDTO, stateChannel, statisticsChannel)
-	if err == context.Canceled {
-		return ErrConnectionCancelled
+	if err != nil {
+		if err == context.Canceled {
+			return ErrConnectionCancelled
+		}
+		manager.sendSessionStatus(dialog, sessionDTO.ID, connectivity.StatusConnectionFailed, err)
```

Comment:
@zolia  Here we will get underlying connection error. As I remember we talked that in case of openvpn conn failure sometimes error is not clear. In order to get openvpn errors I will need to dig into our go-openvpn library. Probably it would be better to work on that in a separate task.